src/System.Reflection.MetadataLoadContext/tests/src/TestUtils/CoreMetadataAssemblyResolver.cs

```diff
@@ -4,18 +4,25 @@
 
 namespace System.Reflection.Tests
 {
+    /// <summary>
+    /// Resolves the core assembly by creating a simulated core assembly.
+    /// </summary>
     public class CoreMetadataAssemblyResolver : MetadataAssemblyResolver
     {
         public CoreMetadataAssemblyResolver() { }
 
-        public override Assembly Resolve(System.Reflection.MetadataLoadContext context, AssemblyName assemblyName)
+        public override Assembly Resolve(MetadataLoadContext context, AssemblyName assemblyName)
         {
-            if (assemblyName.Name == "mscorlib")
+            string name = assemblyName.Name;
+
+            if (name.Equals("mscorlib", StringComparison.OrdinalIgnoreCase) ||
+                name.Equals("System.Private.CoreLib", StringComparison.OrdinalIgnoreCase) ||
+                name.Equals("System.Runtime", StringComparison.OrdinalIgnoreCase) ||
+                name.Equals("System.Runtime.InteropServices", StringComparison.OrdinalIgnoreCase) ||
```

Comment:
Why this one? It is not a core assembly.

Reply:
Update: the COM interop attributes (e.g. DllImportAttribute) are intrinsic\ECMA types and are defined in InteropServices which type forwards to System.Private.CoreLib (for corefx). So by including InteropServices in this test resolver these types are supported as well as other types including GuidAttribute that also type forward to CoreLib.